discord/client.py

```diff
@@ -1133,34 +1136,68 @@ def _check(*args):
 
     # event registration
 
-    def event(self, coro: Coro) -> Coro:
+    def event(self, event: str) -> Coro:
         """A decorator that registers an event to listen to.
 
         You can find more info about the events on the :ref:`documentation below <discord-api-events>`.
 
         The events must be a :ref:`coroutine <coroutine>`, if not, :exc:`TypeError` is raised.
 
+        Parameters
+        ----------
+        event: str
+            The event to subscribe to.
+            See :ref:`documentation below <discord-api-events>` for valid values.
+
         Raises
         ------
         TypeError
             The coroutine passed is not actually a coroutine.
+        ValueError
+            The event does not start with 'on_'
 
         Example
         -------
 
         .. code-block:: python3
 
+            @client.event("on_ready")
+            async def ready_handler():
+                print('Ready!')
+
             @client.event
             async def on_ready():
                 print('Ready!')
         """
 
-        if not asyncio.iscoroutinefunction(coro):
-            raise TypeError("event registered must be a coroutine function")
+        def _event(coro: Coro) -> Coro:
+            """Decorator to register the event."""
+            if not asyncio.iscoroutinefunction(coro):
+                raise TypeError("event registered must be a coroutine function")
```

Comment:
```suggestion
                raise TypeError("Event registered must be a coroutine function")
```
capitalization consistency